### PR TITLE
[WIP] Handle top-level help file with templating

### PR DESCRIPTION
**Pull request by bkabrda**
This is an alternative approach to #33 that we could take to solve #16 - it uses a simple form of templating and is probably more readable than #33 and tries to behave more closely to suggestions by @hhorak in #16.

I think I'm starting to prefer this approach and hence I'd suggest going this way and closing #33, but I'd like to reach consensus before completing this PR and closing #33. Please comment.

### CODE REVIEW

**Comment by bkabrda** [comment]
Obviously, this is far from perfect - e.g. it doesn't check whether the `help.values` files exist etc. I'll fix all similar issues if we decide to pursue this approach.

File: x.y/fedora/Makefile [hunk 1]
```diff
@@ -2,8 +2,16 @@
...
 IMAGE_NAME = image
+SUBSTITUTE_VAR_REGEX = {{\s*[[:alnum:]_]\{1,\}\s*}}
 
 doc:
+	cp ../../root/help.md ./root/
+	source ./root/help.values; \
+	source ../root/help.values; \
+	substitute_vars=`grep "${SUBSTITUTE_VAR_REGEX}" ../../root/help.md  | tr -d "{}[:blank:]"`; \
+	for var in $$substitute_vars; do \
+		sed -i "s|{{\s*$$var\s*}}|$${!var}|" ./root/help.md; \
```
Comment (TomasTomecek) [inline x.y/fedora/Makefile]: Slavku, what do you think about using [envsubst](https://www.gnu.org/software/gettext/manual/html_node/envsubst-Invocation.html) here?

**Review by TomasTomecek — CHANGES_REQUESTED** [review]
I really like this templating approach. Worst case we could even stick to jinja if the present solution would not be sufficient. The only issue for me is the shell script which performs the templating. I would prefer a designated tool for that, such as [envsubst](https://www.gnu.org/software/gettext/manual/html_node/envsubst-Invocation.html).

**Comment by bkabrda** [comment]
Heh, I've actually never heard about envsubst, but you're right that it seems perfect for this kind of thing. Thanks for pointing that out, I'll rework this PR to use it.

**Comment by bkabrda** [comment]
@TomasTomecek I repushed a PR that uses `envsubst` and also adds documentation. If this looks good to you, I'll extend the approach to all the other `Makefile`s so that this can be merged.

**Comment by bkabrda** [comment]
@TomasTomecek I did one further change - replaced the `root` in `root/`, `<version>/root` and `<version>/<os-variant>/root` directories with `help/` for purpose of storing `help.md` and `help.values` (the generated `help.1` will obviously still reside in `root` dir).

Furthermore, I extended this approach to all versions/os-variants, so I believe this is everything needed to be done in this PR. Please review and merge if you think it's ok.

**Comment by TomasTomecek** [comment]
LGTM, well done, Slavku!

**Pull request merged**